adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs

```diff
@@ -172,6 +172,26 @@ public byte[] Serialize()
             }
         }
 
+        /// <summary>
+        /// Serializes current state of the cache as a blob. Caller application can persist the blob and update the state of the cache later by 
+        /// passing that blob back in constructor or by calling method Deserialize.
+        /// </summary>
+        /// <returns>
+        /// A tuple containing the following information:
+        /// <item><see cref="Tuple{T1,T2}.Item1"/>: Current state of the Adal cache as a blob</item>
+        /// <item><see cref="Tuple{T1,T2}.Item2"/>: Current state of the Unified cache as a blob</item>
+        /// </returns>
+        public Tuple<byte[], byte[]> SerializeAdalAndUnifiedCache()
```

Comment:
Do I understand that this is a net new API? (no breaking change)
that customer would need to implement if they care about being compatible?

Reply:
yes if they want to read/write cache in both formats